src/NHapi.Base/Parser/PipeParser.cs

```diff
@@ -61,6 +61,16 @@ static PipeParser()
         /// </summary>
         public PipeParser()
         {
+            ParserConfiguration = new ParserConfiguration();
+        }
+
+        /// <summary>
+        /// Creates a new PipeParser.
+        /// <param name="parserConfiguration">Contains configuration that will be applied when parsing.</param>
+        /// </summary>
+        public PipeParser(ParserConfiguration parserConfiguration)
```

Comment:
@milkshakeuk You mentioned that we should follow the pattern of [newtonsoft](https://www.newtonsoft.com/json/help/html/NamingStrategyCamelCase.htm) or the new [System.Text.JSON](https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-customize-properties?pivots=dotnet-6-0#use-camel-case-for-all-json-property-names) as an example in terms of how nhapi should do "settings" or "options". Does this approach satisfy your intent for settings handling?

_i.e._ You can construct a parser with the desired configuration, and you can also modify this configuration before calling Parse to change behaviour.

Reply:
I feel like this would work yes, I do feel we need the ability to pass in the `ParserConfiguration` to the `Parse` methods (and possibly Encode methods in the futere) so the settings can be specific per call to parse exactly the same as [newtonsoft](https://www.newtonsoft.com/json/help/html/NamingStrategyCamelCase.htm) and [System.Text.Json](https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-customize-properties?pivots=dotnet-6-0#use-camel-case-for-all-json-property-names).

i.e. 
```csharp
// newtonsoft
JsonConvert.SerialiseObject(object, settings);

// System.Text.Json
JsonSerializer.Serialize(object, options);
```
respectively.